ZEngine/include/ZEngine/Rendering/Shaders/Compilers/CompilationStage.h

```diff
@@ -13,18 +16,44 @@ namespace ZEngine::Rendering::Shaders::Compilers
         CompilationStage();
         virtual ~CompilationStage();
 
-        /**
-         * Run Compiler stage
-         *
-         * @param information Collection of shader information
-         */
-        virtual void Run(std::vector<ShaderInformation>& information) override;
-
         /**
          * Run asynchronously compiler stage
          *
          * @param information Collection of shader information
          */
-        virtual std::future<void> RunAsync(std::vector<ShaderInformation>& information) override;
+        virtual std::future<void> RunAsync(ShaderInformation& information) override;
+
+        EShLanguage GetEShLanguage(const ShaderType type);
+
+        void SetShaderRules(glslang::TShader& shader, ShaderInformation& information_list, TBuiltInResource& Resources);
+
     };
+
+    class GlslangInitializer
```

Comment:
not sure I see where you use this class?

Reply:
this is just a RAII implementation of glslang, just like smart pointers, to ensure its finalized every time it is initialized

used here: [CompilationStage.cpp](https://github.com/jnyfah/RendererEngine/blob/b6e7aee04f3897f04e73afd670d5d7eb5298d0ce/ZEngine/src/CompilationStage.cpp#L59-L64)

` GlslangInitializer glslangInit;`
`if (!glslangInit.isSuccess())
   {
        m_information = {false, "Failed to initialize glslang"};
            co_return;
}
`